packages/paste-website/src/pages/patterns/filter/index.mdx

```diff
@@ -0,0 +1,698 @@
+                variant="reset"
+                size="reset"
+                borderRadius="borderRadiusPill"
+              >
+                <FormPill
+                  {...pillState}
+                  selected={isSelected}
+                  onDismiss={
+                    // Remove the filter from the selected filters
+                  }
+                >
+                  {!isSelected ? <PlusIcon decorative /> : null}
+                  <FilterPillView label={filterMap[pill].label} selectedType={isSelected ? pill : null} selectedValue={value} />
+                </FormPill>
+              </PopoverButton>
+              <Popover aria-label={pill} width="size40">
+                <FilterComponent value={value} onApply={onApply} popover={popover} onRemove={onRemove} /> // Create different filter components
+              </Popover>
+            </PopoverContainer>
+          );
+        })}
+      </FormPillGroup>
+      <Box display="flex" justifyContent="space-between" alignItems="center" columnGap="space30" marginTop="space50">
+        <Box display="flex" columnGap="space30">
+          <DetailText marginTop="space0">
+            <Text as="span" color="colorTextWeak" fontSize="fontSize30">
+              {filteredTableData.length} result{filteredTableData.length !== 1 && "s"}
+            </Text>
+          </DetailText>
+          {filteredTableData.length !== data.length ? (
+            <Button variant="link" onClick={handleClearAll}>
+              Clear all
+            </Button>
+          ) : null}
+        </Box>
+      </Box>
+      <DataGrid data={filteredTableData} />
+    </>
+  )
+}`}
+/>
+
+Use it when filter selections naturally lead to additional criteria, or where certain filter formulas are only necessary for very specific use cases.
+
+<DoDont>
+  <Do title="Do" body="Consider the performance implications of loading conditional filters." />
+  <Do title="Do" body="Only one conditional filter should be expanded at a time." />
+  <Dont title="Don't" body="Don’t use conditional filtering for essential filter criteria." />
+</DoDont>
+
+## Behavior
+
+### Fetching results
+
+#### Batch filtering
+
+Requires the user to click an “apply” button to see the results.
+
+ <ResponsiveImage src={FetchingImage} alt="The filter bar supports batch filtering of results on click of apply" />
+
+
+When to use:
+- Use when there are categories with multiple interdependent filter values, and the user might want to take more time selecting the right group of filters.
+- Use when the dataset cannot be refreshed automatically and the system needs time to load the data.
+- Use when you want to prevent 'no results' scenarios.
+
+Loading Results:
+
+Once the user clicks the Apply button, the popover will automatically close, and a loading screen will be displayed until the data is fully loaded. It is recommended to use the [Skeleton Loader](/components/skeleton-loader) when loading the results of the filters and/or search.
```

Comment:
```suggestion
Use batch filtering when:
- There are categories with multiple interdependent filter values, and the user might want to take more time selecting the right group of filters.
- The dataset cannot be refreshed automatically and the system needs time to load the data.
- You want to prevent "no results" scenarios.

To load results: Once the user clicks the "Apply" button, the Popover will automatically close, and a loading screen will be displayed until the data is fully loaded. It is recommended to use the [Skeleton Loader](/components/skeleton-loader) when loading the results of the filters and/or search.
```